lib/ransack/search.rb

```diff
@@ -27,11 +27,12 @@ def result(opts = {})
 
     def build(params)
       collapse_multiparameter_attributes!(params).each do |key, value|
-        case key
-        when 's', 'sorts'
+        if key == 's' || key == 'sorts'
           send("#{key}=", value)
-        else
-          base.send("#{key}=", value) if base.attribute_method?(key)
+        elsif @base.attribute_method?(key)
+          base.send("#{key}=", value)
+        elsif @context.ransackable_scope?(key, @context.object)
```

Comment:
@avit it was a long time ago but maybe you remember why this is implemented in `Search` class? I think it will be more convenient to have this logic in `Grouping` class because it will be possible to build scopes inside groupings, e.g.  params below would be correct:

```
g: [
  { over_age: 20 },
  { name_eq: 'Denis' },
  c: [ ...]
]
```

Regarding to using OR with scopes  it still will be incorrect, but it should not break current behavior of scopes.

Please, correct me if it can't be done on `Grouping` layer for some reason